### ISSUE TITLE
Pipeline: Define all schema data using upickle

### ISSUE THREAD

**Issue by kkasravi** [intel-hadoop/gearpump#686]
Parent task: #400 Kafka -> HBase pipeline

Based on this JSON example

```
{ 
  id: 12394023,
  on: 2015-03-24T06:57:23+00:00,
  body: {
    "source_id": "ip-172-1-2-3",
    "event_id": "c4e6e427-7e45-41d8-9ba2-2223062398dc",
    "event_ts": 1417763522224,
    "metrics": [
        {
            "key": "cpu_load_5min",
            "value": 0.0029296875
        },
        {
            "key": "cpu_load_10min",
            "value": 0.0146484375
        },
        {
            "key": "cpu_load_15min",
            "value": 0.04541015625
        },
        {
            "key": "free_memory",
            "value": 3402670080
        }
    ],
    "event_number": 30
  }
}

JSON schema
{ 
  id: [v4 uuid],  
  on: [UTC timestamp] ,
  body: [inbound message content in UTF-8 encoded string] 
}
```

**Comment by kkasravi** [comment]
Oddly - this seems to hang scalac

```
package org.apache.gearpump.experiments.pipeline

import java.util.Date

object Messages {

  trait Datum[T] {
    val key: String
    val value: T
  }
  case class Cpu(key:String,value:Double) extends Datum[Double]

  case class Memory(key:String,value:Long) extends Datum[Long]

  trait Handler[T] {
    def handle(data: String): Datum[T]
  }
  object Handler {
    import upickle._
    implicit object CpuHandler extends Handler[Double] {
      override def handle(data: String): Cpu = {
        read(data)
      }
    }
    implicit object MemoryHandler extends Handler[Long] {
      override def handle(data: String): Memory = {
        read(data)
      }
    }
  }

  case class Body(sourceId: String, eventId: String, eventTS: Long, metrics: Array[Datum[_]])

  case class Envelope(id: Long, on: Date, body: Body)

  object Converters {
    def convert[T](data: String)(implicit handler: Handler[T]): Datum[T] = {
      handler.handle(data)
    }
  }

}
```

**Comment by manuzhang** [comment]
Hi Kam. uPickle has limitations on trait, generics and data types it can generate.

The following definition would give what you want
```scala
case class Datum(key: String, value: Double)
case class Body(sourceId: String, eventId: String, eventTS: Long, metrics: Array[Datum])
```

**Comment by kkasravi** [comment]
Actually things are working now. I agree I could have made this simpler I am experimenting with typeclass definitions since I think we will want the ability to handle new data types automatically in the framework.

**Comment by kkasravi** [comment]
Closed by #400
